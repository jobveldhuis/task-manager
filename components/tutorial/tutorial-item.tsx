import { IconName } from "@/ui/icon";
import { StyleSheet, View, ViewStyle } from "react-native";
import { Title } from "@/ui/title";
import { Text } from "@/ui/text";
import { FilteredStyleProps } from "@/util/filtered-style-props.type";

type TutorialItemProps = {
  title: string;
  icon: IconName;
  children: string;
  style?: FilteredStyleProps<ViewStyle>;
};

export function TutorialItem({
  title,
  icon,
  children,
  style,
}: TutorialItemProps): JSX.Element {
  return (
    <View style={style}>
      <Title icon={icon} style={styles.title}>
        {title}
      </Title>
      <Text>{children}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  title: {
    marginBottom: 8,
  },
});
