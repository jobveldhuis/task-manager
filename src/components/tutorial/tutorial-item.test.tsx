import { render, screen } from "@testing-library/react-native";
import { TutorialItem } from "./tutorial-item";

describe("TutorialItem", () => {
  const component = (
    <TutorialItem title="Foo is a bar" icon="flag-checkered">
      This is a beautiful time to be alive!
    </TutorialItem>
  );
  it("should match snapshot", () => {
    expect(render(component).toJSON()).toMatchSnapshot();
  });

  it("should contain a title with icon", () => {
    render(component);

    // This container only exists when an icon is rendered with a title.
    expect(screen.getByTestId("title-container")).toBeTruthy();
    expect(screen.getByText("Foo is a bar")).toBeTruthy();
  });

  it("should contain text content", () => {
    render(component);

    expect(
      screen.getByText("This is a beautiful time to be alive!"),
    ).toBeTruthy();
  });
});
