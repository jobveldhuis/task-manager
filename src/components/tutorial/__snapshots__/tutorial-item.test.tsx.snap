// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TutorialItem should match snapshot 1`] = `
<View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 0,
          "flexDirection": "row",
          "gap": 16,
        },
        {
          "marginBottom": 8,
        },
      ]
    }
    testID="title-container"
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "#000C49",
            "fontSize": 24,
            "fontWeight": "bold",
          },
          {
            "color": "#000C49",
            "fontSize": 24,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Foo is a bar
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#000C49",
          "fontSize": 16,
          "fontWeight": "600",
        },
        {
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    This is a beautiful time to be alive!
  </Text>
</View>
`;
